akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala

```diff
@@ -245,12 +243,16 @@ private[remote] object InboundCompression {
     nextTable:               DecompressionTable[T],
     advertisementInProgress: Option[CompressionTable[T]]) {
 
-    def startUsingNextTable(): State[T] =
+    def startUsingNextTable(): State[T] = {
+      // wrap around to positive values
+      val nextVersion = (nextTable.version + 1) & 0x7F
       State(
         oldTable = activeTable,
         activeTable = nextTable,
-        nextTable = DecompressionTable.empty[T].copy(version = nextTable.version + 1),
+        // skip 0 when wrapped around
+        nextTable = DecompressionTable.empty[T].copy(version = (if (nextVersion == 0) 1 else nextVersion).byteValue),
```

Comment:
Very minor nitpick: I'd have liked the entire calculation of next version in one place and then use it here rather than split up on this line and the but up on line 248